src/controls.ts

```diff
@@ -0,0 +1,330 @@
+//@ts-check
+
+import { Vec3 } from 'vec3'
+import { proxy, subscribe } from 'valtio'
+
+import { ControMax } from 'contro-max/build/controMax'
+import { CommandEventArgument, SchemaCommandInput } from 'contro-max/build/types'
+import { stringStartsWith } from 'contro-max/build/stringUtils'
+import { isGameActive, showModal, gameAdditionalState, activeModalStack, hideCurrentModal } from './globalState'
+import { reloadChunks } from './utils'
+
+// doesnt seem to work for now
+const customKeymaps = proxy(JSON.parse(localStorage.keymap || '{}'))
+subscribe(customKeymaps, () => {
+  localStorage.keymap = JSON.parse(customKeymaps)
+})
+
+export const contro = new ControMax({
+  commands: {
+    general: {
+      jump: ['Space', 'A'],
+      inventory: ['KeyE', 'X'],
+      drop: ['KeyQ', 'B'],
+      sneak: ['ShiftLeft', 'Right Stick'],
+      sprint: ['ControlLeft', 'Left Stick'],
+      nextHotbarSlot: [null, 'Left Bumper'],
+      prevHotbarSlot: [null, 'Right Bumper'],
+      attackDestroy: [null, 'Right Trigger'],
+      interactPlace: [null, 'Left Trigger'],
+      chat: [['KeyT', 'Enter'], null],
+      command: ['Slash', null],
+    },
```

Comment:
Why is this in typescript?

Reply:
If there is an enforcement to not use it, it won't be a problem for me to not add the blue bar to your repo. I added it for my personal needs (to speed up development & gain confidence and for some other contribs as well). I should not care about your infra decisions, but it would be great if you could participate and say what you want to be picked. I will also try to get another dev that could probably adopt the infra changes.
That's also the reason why I threw away the idea of rebasing & splitting it into smaller (a bit) prs. I just need to understand what is allowed here and what is not. Again, some kind of task list would be an ideal solution. You can hop to my repo if you want some kind of build preview. Also, please feel free to ask anything, I can explain anything you want.

Reply:
I could barely read more than 500 commits (I believe the commit history could be much cleaner) and I noticed a huge number of various important fixes so I would not call this just a small bag of improvements. That’s why I really I wish it could be available some day (really want to try it out), so @zardoy why did you have to do this unnecessary infra changes, why you couldn’t skip this part!? Also, doesn’t make ts harder to contribute as you need to learn & write types?